### ISSUE TITLE
[method-signature-style] allow skip overload

### ISSUE THREAD

**Issue by loynoir** [typescript-eslint/typescript-eslint#4106]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
    "@typescript-eslint/method-signature-style": [
      "error"
    ],
```

```TS
export interface Foo {
  bar(): string;
  bar(x: number): any;
  bar(x: string): any;
  bar(x: string[]): any;
}
```

**Expected Result**

```ts
export interface Foo {
  bar(): string;
  bar(x: number): any;
  bar(x: string): any;
  bar(x: string[]): any;
}
```

**Actual Result**

`eslint --fix`

Unreadable output 😢
```ts
export interface Foo {
  bar: (() => string) & ((x: number) => any) & ((x: string) => any) & ((x: string[]) => any)
}
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |

**Comment by bradzacher** [comment]
Overloads are still methods. Ignoring them just because you might not like the alternative isn't a good idea because it introduces unsafety into the codebase (which is the opposite of what the rule enforces)

This is the easiest way to fix them.
You can use tools like prettier to format your code which would put them each on a new line, or you can do that manually. ESLint fixers do not care for formatting on purpose because everyone has different styles.

**Comment by JoshuaKGoldberg** [comment]
That's... what the rule does, though? https://github.com/typescript-eslint/typescript-eslint/blob/076d65a/packages/eslint-plugin/docs/rules/method-signature-style.md#options

@loynoir have you read the docs? What's up?

**Comment by loynoir** [comment]
@JoshuaKGoldberg 
Emm, `@typescript-eslint/method-signature-style` doesn't like overload method within class, and union all overload into one very long expression.

Overload is sometime complicated enough. I think readability first in that situation. Thus, suggest `@typescript-eslint/method-signature-style` to allow skiping overload.

Although in this very very simple reproduce,  prettier did some help.
```js
export interface Foo {
  bar: (() => string) &
    ((x: number) => any) &
    ((x: string) => any) &
    ((x: string[]) => any);
}
```

My original case is a function wrapper.  After `prettier` not helpful, I came here.


@bradzacher 
I know it's not recommended, but in some situations I'm willing to take the risk to get more readability...

But, still OK to be closed.
